tasmota/xdsp_15_tm1637.ino

```diff
@@ -994,20 +995,40 @@ void TM1637ShowTime()
       hr = 12;
   }
 
-  char tm[5];
+  char tm[9];
   if (hr < 10)
   {
     if (mn < 10)
-      snprintf(tm, sizeof(tm), PSTR("%c%d0%d"), z, hr, mn);
+    {
+      if (sc < 10)
+        snprintf(tm, sizeof(tm), PSTR("%c%d0%d0%d"), z, hr, mn, sc);
+      else
+        snprintf(tm, sizeof(tm), PSTR("%c%d0%d%d"), z, hr, mn, sc);
+    }
     else
```

Comment:
Why these strange construct for values below 10???

I suggest you take a look at this option:
```
snprintf_P(tm, sizeof(tm), PSTR("%c%d%02d%02d"), z, hr, mn, sc);
```

Reply:
Fixed. Thanks.